### PR TITLE
Adds custom prompt api

### PR DESCRIPTION
**Pull request by danjm**
This PR branches off of #80.

This adds a new api that plugins can use to trigger custom prompts within the metamask ui. It supports the plugin added with: https://github.com/MetaMask/mm-plugin/pull/34

### CODE REVIEW

File: app/scripts/controllers/accounts.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+    const handler = this.pluginsController.accountMessageHandlers.get(uniquePluginAccountDomains[0])
+    if (uniquePluginAccountDomains.length === 0 || !handler) {
+      throw new Error(`No handler plugin for account ${address} found.`)
+    }
+
+    return handler
+  }
+
+  getOrigin (address) {
```
Comment (danfinlay) [inline app/scripts/controllers/accounts.js]: This function is being called without `this.` prefix throughout the file. A hacker at the DeFi hackathon found the issue and will be pushing the fix upstream!

Comment (tommyz7) [reply app/scripts/controllers/accounts.js]: Fix in #90

File: app/scripts/controllers/permissions/restrictedMethods.js [hunk 2]
```diff
@@ -138,11 +138,11 @@ function getExternalRestrictedMethods (permissionsController) {
 
     'prompt': {
```
Comment (danfinlay) [inline app/scripts/controllers/permissions/restrictedMethods.js:138]: Break into `customPrompt`, to preserve backwards-compat.

**Comment by rekmarks** [comment]
Superseded by #150

**Comment by Ricardo11v** [comment]
Hola

**Pull request closed without merging**